### PR TITLE
Sol blockhash tests

### PR DESCRIPTION
**Pull request by LefterisJP**
A small PR with test for the blockhash command in Solidity.

Also some minor improvements to `SolidityExecutionFramework.h`

### CODE REVIEW

File: test/solidityExecutionFramework.h [hunk 1]
```diff
@@ -46,7 +46,20 @@ class ExecutionFramework
...
-		compiler.compile(_sourceCode, m_optimize);
+		try
+		{
+			compiler.compile(_sourceCode, m_optimize);
+		}
+		catch (const std::exception& e)
+		{
+			std::string const* extra = boost::get_error_info<errinfo_comment>(e);
+			std::string msg = std::string("Parsing contract failed with: ") +
+				e.what() + std::string("\n");
```
Comment (CJentzsch) [inline test/solidityExecutionFramework.h]: you could just do "msg =std::string("Parsing contract failed with: ") + diagnostic_information(_e);" Then you get all possible information.

File: test/solidityExecutionFramework.h [hunk 2]
```diff
@@ -46,7 +46,20 @@ class ExecutionFramework
 	bytes const& compileAndRun(std::string const& _sourceCode, u256 const& _value = 0, std::string const& _contractName = "")
 	{
 		dev::solidity::CompilerStack compiler;
-		compiler.compile(_sourceCode, m_optimize);
+		try
+		{
+			compiler.compile(_sourceCode, m_optimize);
+		}
+		catch (const std::exception& e)
```
Comment (CJentzsch) [inline test/solidityExecutionFramework.h]: _e

File: test/solidityExecutionFramework.h (hunk 1 above)
Comment (LefterisJP) [inline test/solidityExecutionFramework.h]: very good point. Will adjust here and in 2 other places in tests that I can remember.

File: test/solidityExecutionFramework.h (hunk 2 above)
Comment (LefterisJP) [inline test/solidityExecutionFramework.h]: did not see that in the coding standard. But by checking around the code we seem to always put underscore in the exception in catch so I will comply. Perhaps we should add a line in the coding standard? A quick look did not show anything.

**Comment by LefterisJP** [comment]
Ok will do, but this is working too since boost::exceptions itself also derives from std::exception. But yes it's better to let those exceptions propagate upwards.

**Pull request merged**